src/message-processor/message-processor.ts

```diff
@@ -1,25 +1,114 @@
-import logger from "../utils/logger";
 import { Metrics, MetricUnit } from "@aws-lambda-powertools/metrics";
-import { SQSEvent } from "aws-lambda";
+import { SQSEvent, SQSRecord } from "aws-lambda";
+
+import { MetricDimension, MetricName } from "../types/metricEnum";
+import { isTxmaMessage, TxmaMessage } from "../types/txmaMessage";
 
-import { MetricEnum } from "../types/metricEnum";
-import { MessageProcessorRequest } from "../types/messageProcessorRequest";
+import { getSecret } from "@aws-lambda-powertools/parameters/secrets";
+import { getConfiguration, type Configuration } from "../types/configuration";
+import { getString } from "../types/stringutils";
+import logger from "../utils/logger";
+import { isErrorResponse } from "../types/endpoint";
 
-const messageProcessorMetrics = new Metrics();
+const metrics = new Metrics();
 
+/**
+ * AWS Lambda handler
+ * @param event Simple Queue Service event
+ */
 export const handler = async (event: SQSEvent): Promise<void> => {
-  for (const record of event.Records) {
-    try {
-      const request = JSON.parse(record.body) as MessageProcessorRequest;
-      logger.info("Message body", { logIds: request.logIds });
-      messageProcessorMetrics.addMetric(MetricEnum.MessagesReceived, MetricUnit.Count, 1);
-    } catch (e) {
-      logger.error("Failed to process record", {
-        error: e instanceof Error ? e : { message: String(e) },
-        rawBody: record.body,
-      });
+  const records = parseSQSRecords(event.Records);
+  const apiKey: string | undefined = await getServiceApiKey();
+
+  metrics.addMetric(MetricName.MessagesReceived, MetricUnit.Count, records.length);
+
+  const config = await getConfiguration();
+
+  try {
+    for (const record of records) {
+      if (!isInterventionRecord(record, config)) {
+        continue;
+      }
+
+      await invalidateUser(record.user_id, record.intervention_code, config.evcsApiUrl, apiKey);
     }
+  } finally {
+    metrics.publishStoredMetrics();
   }
+};
 
-  messageProcessorMetrics.publishStoredMetrics();
+/**
+ * Invalid the given user id
+ * @param userId The user id to invalidate
+ */
+const invalidateUser = async (userId: string, interventionCode: string, baseUrl: string, apiKey?: string) => {
```

Comment:
Please can you add `interventionCode`, `baseUrl` and `apiKey` to the the JSDoc? There are currently warnings about this in the IDE.